File: support-products/src/main/java/com/gabriel/products/core/domain/model/Category.java
package com.gabriel.products.core.domain.model;

public enum Category {

    BURGER("burger"),

    ACCOMPANIMENT("accompaniment"),

    DESSERT("dessert"),

    DRINK("drink");

    private final String value;

    Category(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }
}

File: menu/internal/domain/category.go
package domain

import "fmt"

type Category string

const (
	CategoryBurger        Category = "burger"
	CategoryAccompaniment Category = "accompaniment"
	CategoryDessert       Category = "dessert"
	CategoryDrink         Category = "drink"
)

var validCategories = map[Category]bool{
	CategoryBurger: true, CategoryAccompaniment: true,
	CategoryDessert: true, CategoryDrink: true,
}

func ParseCategory(s string) (Category, error) {
	c := Category(s)
	if !validCategories[c] {
		return "", fmt.Errorf("invalid category: %s", s)
	}
	return c, nil
}

func (c Category) String() string { return string(c) }
